app/controllers/renewal/WhatYouNeedController.scala

```diff
@@ -15,29 +15,45 @@
  */
 
 package controllers.renewal
-
+import cats.data.OptionT
+import cats.implicits._
+import connectors.DataCacheConnector
 import controllers.{AmlsBaseController, CommonPlayDependencies}
 import javax.inject.{Inject, Singleton}
+import models.businessmatching.{BusinessActivities, BusinessMatching, BusinessMatchingMsbServices}
 import models.registrationprogress.{NotStarted, Section, Started}
-import play.api.mvc.MessagesControllerComponents
+import play.api.mvc.{MessagesControllerComponents, Request}
 import services.RenewalService
 import utils.AuthAction
 import views.html.renewal._
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
-
 @Singleton
-class WhatYouNeedController @Inject()(val authAction: AuthAction,
-                                      val ds: CommonPlayDependencies,
-                                      renewalService: RenewalService,
-                                      val cc: MessagesControllerComponents)(implicit ec: ExecutionContext) extends AmlsBaseController(ds, cc) {
+class WhatYouNeedController @Inject()(
+                                       val dataCacheConnector: DataCacheConnector,
+                                       val authAction: AuthAction,
+                                       val ds: CommonPlayDependencies,
+                                       renewalService: RenewalService,
+                                       val cc: MessagesControllerComponents)(implicit ec: ExecutionContext) extends AmlsBaseController(ds, cc) {
 
   def get = authAction.async {
     implicit request =>
-      renewalService.getSection(request.credId) map {
-        case Section(_,NotStarted | Started,_,_) => Ok(what_you_need())
-        case _ => Redirect(controllers.routes.RegistrationProgressController.get())
-      }
+        (for {
+          bm <- OptionT(dataCacheConnector.fetch[BusinessMatching](request.credId, BusinessMatching.key))
+          ba <- OptionT.fromOption[Future](bm.activities)
+          section <- OptionT.liftF(getSection(renewalService, request.credId, Some(ba), bm.msbServices))
+        } yield {
+          section
+        }).getOrElse(InternalServerError("Unable to retrieve the correct data"))
```

Comment:
Why has the error message changed here? Should this not be "Unable to retrieve business activities" for consistency?

Reply:
Done